File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/interactive/InteractiveUseRequestMessage.scala





















// Generated on 05/03/2013 10:34:52
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait InteractiveUseRequestMessage extends MessageDeserializer {
  val messageId = 5001

  protected trait Message extends BaseMessage with Serializable {
    val messageId = InteractiveUseRequestMessage.messageId

    val elemId: Int
    val skillInstanceUid: Int

    def serialize(buf: Buffer) {
      buf.writeInt(elemId)
      buf.writeInt(skillInstanceUid)
    }

  }

  protected def create(elemId: Int, skillInstanceUid: Int): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val elemId = buf.readInt()
    val skillInstanceUid = buf.readInt()

    create(elemId, skillInstanceUid)
  }

}

object InteractiveUseRequestMessage extends InteractiveUseRequestMessage {
  case class MessageImpl(elemId: Int, skillInstanceUid: Int) extends Message
  type Target = MessageImpl

  override def create(elemId: Int, skillInstanceUid: Int) = MessageImpl(elemId, skillInstanceUid)

  def apply(elemId: Int, skillInstanceUid: Int) = create(elemId, skillInstanceUid)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/InteractiveUseRequestMessage.java


// Generated on 05/08/2013 19:37:52
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class InteractiveUseRequestMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5001;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int elemId;
    public int skillInstanceUid;
    
    public InteractiveUseRequestMessage() { }
    
    public InteractiveUseRequestMessage(int elemId, int skillInstanceUid) {
        this.elemId = elemId;
        this.skillInstanceUid = skillInstanceUid;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeInt(elemId);
        buf.writeInt(skillInstanceUid);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        elemId = buf.readInt();
        if (elemId < 0)
            throw new RuntimeException("Forbidden value on elemId = " + elemId + ", it doesn't respect the following condition : elemId < 0");
        skillInstanceUid = buf.readInt();
        if (skillInstanceUid < 0)
            throw new RuntimeException("Forbidden value on skillInstanceUid = " + skillInstanceUid + ", it doesn't respect the following condition : skillInstanceUid < 0");
    }
    
}
